ee/hogai/tools/subagent.py

```diff
@@ -0,0 +1,218 @@
+import uuid
+from typing import Literal, Self, cast
+
+import structlog
+from langchain_core.runnables import RunnableConfig
+from pydantic import BaseModel, Field
+
+from posthog.schema import (
+    ArtifactContentType,
+    ArtifactMessage,
+    AssistantEventType,
+    AssistantGenerationStatusEvent,
+    AssistantMessage,
+    AssistantTool,
+    AssistantToolCallMessage,
+    AssistantUpdateEvent,
+    HumanMessage,
+)
+
+from posthog.models import Team, User
+
+from ee.hogai.context.context import AssistantContextManager
+from ee.hogai.core.executor import AgentExecutor
+from ee.hogai.stream.redis_stream import get_subagent_stream_key
+from ee.hogai.tool import MaxTool, ToolMessagesArtifact
+from ee.hogai.utils.prompt import format_prompt_string
+from ee.hogai.utils.types.base import AgentType, ArtifactRefMessage, AssistantMessageUnion, AssistantState, NodePath
+from ee.models import Conversation
+
+logger = structlog.get_logger(__name__)
+
+
+SUBAGENT_TOOL_PROMPT = """
+Use this tool to spin up a dedicated agent that can independently tackle complex, multi-step workflows.
+
+# Available agents and modes they can switch between:
+{{agents_prompt}}
+
+{{modes_prompt}}
```

Comment:
Mustache escapes similarly to HTML, which we don't need. Triple curly brackets disable it.

```suggestion
{{{agents_prompt}}}

{{{modes_prompt}}}
```